plutus-core/plutus-ir/src/PlutusIR/Compiler.hs

```diff
@@ -45,13 +45,18 @@ import           Control.Monad
 import           Control.Monad.Reader
 import           PlutusPrelude
 
--- | Actual simplifier
-simplify :: M.ToBuiltinMeaning uni fun => Term TyName Name uni fun b -> Term TyName Name uni fun b
-simplify = DeadCode.removeDeadBindings . Inline.inline . Beta.beta
+-- | Run the simplifier @n@ times
+simplify' :: forall uni fun b. M.ToBuiltinMeaning uni fun => Integer -> Term TyName Name uni fun b -> Term TyName Name uni fun b
+simplify' n t | n <= 0    = t
+              | otherwise = simplify' (n - 1) (simplify t)
+    where
+        -- | Actual simplifier
+        simplify :: Term TyName Name uni fun b -> Term TyName Name uni fun b
+        simplify = DeadCode.removeDeadBindings . Inline.inline . Beta.beta
 
 -- | Perform some simplification of a 'Term'.
 simplifyTerm :: Compiling m e uni fun a => Term TyName Name uni fun b -> m (Term TyName Name uni fun b)
-simplifyTerm = PLC.rename >=> (runIfOpts $ pure . simplify . simplify . simplify . simplify)
+simplifyTerm = PLC.rename >=> (runIfOpts $ pure . simplify' 4)
```

Comment:
I was thinking we could put this in the compilation options, then we could potentially set it as a plugin flag etc. A bit like GHC's `max-simplifier-iterations`.

Reply:
Yeah, that sounds reasonable.  Do you mind pointing towards the record for compilation options?

Reply:
https://github.com/input-output-hk/plutus/blob/master/plutus-core/plutus-ir/src/PlutusIR/Compiler/Types.hs#L43

Reply:
And I presume you would want to expose it via:
https://github.com/input-output-hk/plutus/blob/jakzale/inline-single-occurrence/plutus-tx-plugin/src/PlutusTx/Plugin.hs#L55-L62

Reply:
Done in 66d171c7c0e0a886d716ad02b2330d86770ec73e